libmariadb/mariadb_lib.c

```diff
@@ -3796,6 +3796,8 @@ int STDCALL mysql_set_character_set(MYSQL *mysql, const char *csname)
 
   if (!csname)
     goto error;
+  if (!strcasecmp("utf8",csname))
+    csname= "utf8mb4";
```

Comment:
I wonder why In ma_charset you set character to utf8mb3 (if is "utf8"), but here you set it to uf8mb4?!

Reply:
Typo, sorry. Shall fix.